### PR TITLE
set number of parallel jobs; fix #23

### LINKED ISSUE #23: make number of parallel jobs an argument

### PR DESCRIPTION
**Pull request by hahahannes**

### CODE REVIEW

**Review by PhilippKaniuth — CHANGES_REQUESTED** [review]
You declare the parameter's type to be `int`, but the option "max" can only by assigned as a string. This would lead to a TypeError later when checking `parallel > 1`. 

I think either declare the parameter to be a `string` or treat some very large integer as the max-option.

File: frrsa/fitting/crossvalidation.py [hunk 1]
```diff
@@ -292,8 +292,8 @@ def start_outer_cross_validation(n_conditions,
     n_outer_cvs : int
         Denotes how many outer crossvalidations are conducted in total.
     parallel : bool
```
Comment (PhilippKaniuth) [inline frrsa/fitting/crossvalidation.py:294]: Also, could you escalate the chaging of the type? :)

File: frrsa/fitting/crossvalidation.py [hunk 2]
```diff
@@ -92,9 +92,9 @@ def frrsa(target,
     predictions_wanted : bool, optional
         Indication of whether predicticted dissimilarities for all outer
         cross-validations shall be returned (defaults to `False`).
-    parallel : bool, optional
-        Indication of whether to parallelize the outer cross-validation,
-        using all of the machine's CPUs cores (defaults to `False`).
+    parallel : int, optional
```
Comment (PhilippKaniuth) [inline frrsa/fitting/crossvalidation.py:95]: Type should be `str` or change the "max" option to large integer.

File: frrsa/fitting/crossvalidation.py [hunk 3]
```diff
@@ -345,15 +345,18 @@ def start_outer_cross_validation(n_conditions,
     results = []
     outer_loop_count = -1
 
-    if parallel:
+    if parallel > 1:
```
Comment (PhilippKaniuth) [inline frrsa/fitting/crossvalidation.py:348]: This would throw an error if `parallel="max"`

**Comment by hahahannes** [comment]
ah yes, thanks for spotting! that implementation was bit rushed @PhilippKaniuth

**Comment by PhilippKaniuth** [comment]
Thanks! But line 350 would still throw an error, right? It would try to check whether the string is bigger than 1. This would throw a TypeError as the operand `>` is not supported between `str` and `int`.

And in 297 `parallel` is still declared and int? Default value of `parallel` in `frrsa` is also an int. Or am I being thick here?:D

**Comment by hahahannes** [comment]
I moved the conversion to line 190. Then  start_outer_cross_validation expects only integers (no max anymore). Thought this might be cleaner instead of str/int type checking in start_outer_cross_validation

**Comment by PhilippKaniuth** [comment]
Apologies, I didn't realize that, that's nice.

However, we can't unambiguously indicate the type then, right? `str` is wrong if the user doesn't input `"max"`, and `int` is wrong if s/he does input `"max"`. And if the user inputs say 2 as a string (since we declared it to be input as a string) there will be the TypeError.

While I agree that the conversion is 190 is nice, I think I would favor making it as easy as possible for the user; which would be to unambiguously indicate the type as `str` and type check it and/or transform it once. What do you think? It could be done easily be amending the if-statement in 190:

```python
if parallel == 'max':
  parallel = psutil.cpu_count(logical=False)
else:
  parallel = int(parallel)
  ```

**Comment by hahahannes** [comment]
aggree! otherwise the documentation would be wrong

**Review by PhilippKaniuth — APPROVED** [review]

**Pull request merged**